frontend/src/components/SystemsTable/index.tsx

```diff
@@ -1,46 +1,51 @@
-import React, { useEffect, useState } from "react";
+import React, { useCallback, useEffect, useState } from "react";
 import "./index.css";
 import { message, Space } from "antd";
 import { SystemSubmitDrawer, AnalysisDrawer } from "../../components";
 import { backendClient, parseBackendError } from "../../clients";
 import { SystemModel, newSystemModel } from "../../models";
-import { Filter, SystemTableTools } from "./SystemTableTools";
+import { SystemTableTools } from "./SystemTableTools";
 import { SystemTableContent } from "./SystemTableContent";
 import { findTask, PageState } from "../../utils";
 import { TaskCategory } from "../../clients/openapi";
 import { LoginState, useUser } from "../useUser";
-
-interface Props {
-  /**initial value for task filter */
-  name?: string;
-  initialTaskFilter?: string;
-  dataset?: string;
-  subdataset?: string;
-  datasetSplit?: string;
-}
+import { useHistory } from "react-router-dom";
+import useQuery from "../useQuery";
+import { FilterUpdate, SystemFilter } from "./SystemFilter";
 
 /** A table that lists all systems */
-export function SystemsTable({
-  name,
-  initialTaskFilter,
-  dataset,
-  subdataset,
-  datasetSplit,
-}: Props) {
+export function SystemsTable() {
   const [pageState, setPageState] = useState(PageState.loading);
   const [page, setPage] = useState(0);
   const [pageSize, setPageSize] = useState(10);
+  const history = useHistory();
 
   const [systems, setSystems] = useState<SystemModel[]>([]);
   const [total, setTotal] = useState(0);
+  const query = useQuery();
+  const [filters, setFilters] = useState<SystemFilter>(
+    SystemFilter.parseQueryToFilter(query)
+  );
 
-  // filters
-  const [nameFilter, setNameFilter] = useState(name);
-  const [taskFilter, setTaskFilter] = useState(initialTaskFilter);
-  const [showMine, setShowMine] = useState(false);
-  const [sortField, setSortField] = useState("created_at");
-  const [sortDir, setSortDir] = useState<"asc" | "desc">("desc");
-  const [split, setSplit] = useState<string | undefined>(datasetSplit);
+  const [activeSystemIDs, setActiveSystemIds] = useState<string[]>([]);
+
+  useEffect(() => {
+    const prevString = query.toString();
+    const newString = filters.toUrlParams().toString();
+    if (prevString !== newString) {
+      history.push({ search: filters.toUrlParams().toString() });
+    }
+  }, [history, filters, query]);
+
+  const updateFilterAndQuery = useCallback(
```

Comment:
This function can be moved into `onFilterChange()` to avoid the use of `useCallback`. Also, if we modify `update()` as I recommended above, I think we can avoid this function.